minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java

```diff
@@ -257,7 +257,9 @@ protected static void writeNiFiProperties(ConfigSchema configSchema, OutputStrea
             orderedProperties.setProperty("nifi.web.jetty.working.directory", "./work/jetty");
             orderedProperties.setProperty("nifi.web.jetty.threads", "200");
 
-            orderedProperties.setProperty("nifi.sensitive.props.key", sensitiveProperties.getKey(), System.lineSeparator() + "# security properties #");
+            final String sensitivePropertiesKey = sensitiveProperties.getKey();
+            final String notnullSensitivePropertiesKey = sensitivePropertiesKey != null ? sensitivePropertiesKey : "";
```

Comment:
Can you add a unit test for this? I think you can use a copy of the bootstrap.conf from minifi-bootstrap/src/test/resources/MINIFI-516 but just set the keystoreType property mentioned in the Jira. It might result in a later error, but if it's not an NPE it could be expected.

Reply:
yep, will do!

Reply:
@mattyb149 I did both. I modified ConfigTransformerTest.checkSSLOverrides() per your suggestion, and also added a new unit test that would have triggered the NPE before